### PR TITLE
Onboarding: recommendations trending section

### PR DESCRIPTION
**Pull request by mchowning**
| 📘 Project: #524  |
| --- |

## Description
Add the trending section to the recommendation screen

## Testing Instructions
1. Enable onboarding in `base.gradle` file. You may want to consider updating the startDestination [here](https://github.com/Automattic/pocket-casts-android/blob/2c6b39dccba6eb67aa643e17e59fdc3df59db437/modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingFlowComposable.kt#L49) to be `OnboardingRecommendationsFlow.route`.
2. Open the onboarding recommendations screen
3. Observe that you can scroll through the trending podcasts for your region
4. Observe that tapping on any of the podcast subscribes/unsubscribes to that podcast

## Screenshots or Screencast 

https://user-images.githubusercontent.com/4656348/206279048-5405ab23-c988-4639-b519-8c4109f2d8de.mov

## Checklist
- [X] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] I have considered whether it makes sense to add tests for my changes
- [X] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [X] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [X] with different themes
- [X] with a landscape orientation
- [X] with the device set to have a large display and font size
- [X] for accessibility with TalkBack

### CODE REVIEW

**Review by ashiagr — APPROVED** [review]
This is amazing, great job @mchowning! 👏 

Apart from the region code comment, I have a couple more findings:

1. Even after selecting a few podcasts from the trending section, I see that the only option I have to proceed with is "Not Now". It might make sense to rename it to "Continue" button.


2. Nitpick: The subscribed state image might look better with a green checkmark similar to how it is when we selected podcasts to share:

Subscribed State | Selected State
-----|-----
 | 

3. After disabling, enabling the internet, and retapping the search bar, I kept seeing no network error and could not go to the search screen.

Other than this, everything worked like a charm! 🥳

File: modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/OnboardingRecommendationsStartPageViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        viewModelScope.launch {
+
+            val feed = try {
+                repository.getDiscoverFeed().await()
+            } catch (e: Exception) {
+                Timber.e(e)
+                return@launch
+            }
+
+            val regionCode = settings.getDiscoveryCountryCode()
```
[the remark this answers is not in the document]
Comment (mchowning) [reply modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/OnboardingRecommendationsStartPageViewModel.kt:45]: That's weird, I tested both German and Australian English and region code was "de" and "au" for me, respectively. 

<details>
<summary>Debug breakpoint screenshot</summary>

Screenshot 2022-12-08 at 10 04 32 AM

</details>

Here's a video of the steps I'm using. Is there something we're doing differently?

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/4656348/206482006-de40245c-bc20-43e2-8d58-2fc8a98dba02.mov

</details>

**Comment by mchowning** [comment]
Thanks for the careful review @ashiagr ! 

> 1. Even after selecting a few podcasts from the trending section, I see that the only option I have to proceed with is "Not Now". It might make sense to rename it to "Continue" button.

You're right. That would be confusing. I made that change in [28e1da0](https://github.com/Automattic/pocket-casts-android/pull/640/commits/28e1da075860f007d10eef33f742119eda816980).

> 2. Nitpick: The subscribed state image might look better with a green checkmark similar to how it is when we selected podcasts to share:

I had a similar thought and actually tried it with a green checkmark. In the end, I decided to stay consistent with how we handle subscribed podcasts in the Discover Feed since that is the other place we have a similar UX of tapping a podcast cover to subscribe. I'll leave this as-is for now (unless you feel strongly, in which case I'm happy to go ahead and make the change), but let me ping @adamzelinski for his input.

<details>
<summary>Screenshot of podcast subscriptions in the discover feed</summary>

Screenshot 2022-12-08 at 10 21 58 AM

</details>

> 3. After disabling, enabling the internet, and retapping the search bar, I kept seeing no network error and could not go to the search screen.

Good catch! Addressed in [08d83bd](https://github.com/Automattic/pocket-casts-android/pull/640/commits/08d83bd7695d7e698d4b30fda53a96250b0da77e).

**Comment by adamzelinski** [comment]
This is looking really great thanks, everyone. 
> I had a similar thought and actually tried it with a green checkmark. In the end, I decided to stay consistent with how we handle subscribed podcasts in the Discover Feed since that is the other place we have a similar UX of tapping a podcast cover to subscribe.
Let's leave it, for now, to stay consistent with the discover feed. I'll make a note to relook at this in the future.

**Pull request merged**